### PR TITLE
Fixed 'Bug 41388 - Code completion is incorrect for array types'

### PR DESCRIPTION
**Pull request by mkrueger**
A very special case which shows a little bug in type inference. Fixed
with a work around.

### CODE REVIEW

File: main/tests/MonoDevelop.CSharpBinding.Tests/Features/CodeCompletion/NR6/MiscTests.cs [hunk 1]
```diff
@@ -210,5 +210,40 @@ public static void Main(string[] args)
...
+{
+	public event EventHandler FooBar;
+
+	public Test[] test { get; private set; }
+
+
+	public Test()
+	{
+		test = new $$
+		FooBar += Test_FooBar;
```
Comment (alanmcgovern) [inline main/tests/MonoDevelop.CSharpBinding.Tests/Features/CodeCompletion/NR6/MiscTests.cs]: Heh, now i see the edge case i was hitting!

Comment (mkrueger) [inline main/tests/MonoDevelop.CSharpBinding.Tests/Features/CodeCompletion/NR6/MiscTests.cs]: first I thought : wtf what's happening in your screencast - but then I saw what's happening - was very funny :). 
In general the roslyn parser is awesome understanding incorrect code :). But there are some cases that are tricky.
